app/README.md

```diff
@@ -0,0 +1,221 @@
+and running and listening on the websocket-ports.
+* `testOut` prints the first argument if `DBG_TEST` is greater or equal to `1`
+* `testGrep` searches the first argument in the output of the given command
+
+The whole list of all `test`-commands can be found [here](#test-commands).
+
+### Helper methods
+
+Each `test.sh` will have its own helper methods, but the most common one is
+to write something to run your CLI-app:
+
+```bash
+runCl(){
+    dbgRun ./$APP -d $DBG_APP $@
+}
+```
+
+This will call your app with the given debugging-option, referenced from the top
+of your `test.sh` for easy change in case you need to debug your test.
+
+Sometimes you might want to give more option, most often the configuration-directory
+to be used:
+
+```bash
+runCl(){
+	local CFG=cl$1
+	shift
+	dbgRun ./$APP -d $DBG_APP -c $CFG $@
+}
+```
+
+This passes the configuration-directory to the app, supposing the app has a
+`-c` argument for passing it.
+
+### A call to main
+
+This is very simple:
+
+```bash
+main
+```
+
+As bash is an interpreter, it needs to run through all your methods before
+being able to call them. And because it is nice to have all the important
+methods at the top, we propose this setup.
+
+## Test-commands
+
+Here is an overview of the currently supported test-commands in `libtest.sh`:
+
+| Command | Arguments | Description |
+| ------- | --------- | ----------- |
+| `test` | `Name` | Prints `Name`, cleans up build-directory, deletes all databases
+from services in previous run, and calls `testName` |
+| `testOut` | `$@` | Outputs all arguments if `DBT_TEST -ge 1` |
+| `dbgRun` | `$@` | Runs `$@` and outputs the result of `$@` if `DBG_TEST -ge 2`.
+Redirects the output in all cases if `OUTFILE` is set. |
+| `testOK` | `$@` | Asserts that the exit-code of  running `$@` using `dbgRun` is `0`. |
+| `testFail` | `$@` | Asserts that the exit-code of  running `$@` using `dbgRun` is NOT `0`. |
+| `testFile` | `File` | Asserts `File` exists and is a file. |
+| `testNFile` | `File` | Asserts that `File` DOES NOT exist. |
+| `testFileGrep` | `String` `File` | Asserts that `String` exists in `File`. |
+| `testGrep` | `String` `$@[1..]` | Asserts that `String` is in the output of the
+command being run by `dbgRun` and all but the first input argument. Ignores
+the exit-code of the command. |
+| `testNGrep` | `String` `$@[1..]` | Asserts that `String` is NOT in the output of the
+command being run by `dbgRun` and all but the first input argument. Ignores
+the exit-code of the command. |
+| `testReGrep` | `String` | Asserts `String` is part of the last command being run
+by `testGrep` or `testNGrep`. |
+| `testReNGrep` | `String` | Asserts `String` is NOT part of the last command being run
+by `testGrep` or `testNGrep`. |
+| `testCount` | `Count` `String` `$@[2..]` | Asserts that `String` exists exactly
+`Count` times in the output of the command being run by `dbgRun` and all but
+the first two arguments. |
```

Comment:
It's easy to forget to update these, so I feel these should go into the code comments first.